hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/protocol/ScmBlockLocationProtocol.java

```diff
@@ -59,8 +60,29 @@
    * @return allocated block accessing info (key, pipeline).
    * @throws IOException
    */
-  List<AllocatedBlock> allocateBlock(long size, int numBlocks,
+  @Deprecated
+  default List<AllocatedBlock> allocateBlock(long size, int numBlocks,
       ReplicationType type, ReplicationFactor factor, String owner,
+      ExcludeList excludeList) throws IOException {
+    return allocateBlock(size, numBlocks, ReplicationConfig
+        .fromTypeAndFactor(type, factor), owner, excludeList);
+  }
+
+  /**
+   * Asks SCM where a block should be allocated. SCM responds with the
+   * set of datanodes that should be used creating this block.
+   *
+   * @param size              - size of the block.
+   * @param numBlocks         - number of blocks.
+   * @param replicationConfig - replicationConfiguration
+   * @param excludeList       List of datanodes/containers to exclude during
```

Comment:
Nit: Add owner to param spec - note it was missing before these changes, so not caused by this change.